src/main/java/seedu/address/model/person/Date.java

```diff
@@ -0,0 +1,54 @@
+package seedu.address.model.person;
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+
+/**
+ * Represents a Person's date in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidDate(String)}
+ */
+public class Date {
+    public static final String MESSAGE_DATE_CONSTRAINTS =
+            "Date can only contain numbers, and should be 6 digits long in the format DDMMYY";
+    public static final String DATE_VALIDATION_REGEX = "\\d{6,}";
```

Comment:
DATE_VALIDATION_REGEX should check if the date is valid, i.e. day and month are in bound, instead of just 6 digits.